src/solvers/dg_p4est/containers_2d.jl

```diff
@@ -20,12 +20,20 @@ end
 # Interpolate tree_node_coordinates to each quadrant
 function calc_node_coordinates!(node_coordinates,
                                 mesh::P4estMesh{2},
-                                nodes)
+                                basis::LobattoLegendreBasis)
```

Comment:
Nothing other than the basis' nodes is used here, right? This function will be called by Trixi2Vtk where we only have the nodes and no basis. I should've commented that, sorry!

Reply:
Would you mind changing that part in Trixi2Vtk? I would prefer passing the basis for consistency with other meshes and to enable multiple dispatch. For example, other meshes use something like
```julia
  # Get cell length in reference mesh
  dx = 2 / size(mesh, 1)
  dy = 2 / size(mesh, 2)

  # Calculate node coordinates of reference mesh
  cell_x_offset = -1 + (cell_x-1) * dx + dx/2
  cell_y_offset = -1 + (cell_y-1) * dy + dy/2
```
which is specific to the `nodes` of the `LobattoLegendreBasis`. In particular, it won't work for FD methods implemented in #617.

What do you think, @sloede?

Reply:
I have no strong feelings either way. However, I'd suggest to follow the principle of not implementing something for a feature that isn't even there yet (YAGNI). Thus if changing this for dispatch is only needed for a future version of FD-SBP, I wouldn't change the interface right now.

On the other hand, if it would remove currently existing features if we continue to use plain nodes instead of passing the `basis`, then we should use the implementation proposed by Hendrik.

Reply:
Well, FD-SBP are already in `main`, so I would like to keep this version here